daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/CompiledExpression.scala

```diff
@@ -34,10 +34,10 @@ import org.apache.daffodil.runtime1.dsom._
 import org.apache.daffodil.runtime1.infoset.DataValue.DataValuePrimitive
 
 object ExpressionCompilers extends ExpressionCompilerClass {
-  override val String = new ExpressionCompiler[String]
-  override val JLong = new ExpressionCompiler[JLong]
-  override val AnyRef = new ExpressionCompiler[AnyRef]
-  override val JBoolean = new ExpressionCompiler[JBoolean]
+  override val String: ExpressionCompiler[String] = new ExpressionCompiler[String]
+  override val JLong: ExpressionCompiler[JLong] = new ExpressionCompiler[JLong]
+  override val AnyRef: ExpressionCompiler[AnyRef] = new ExpressionCompiler[AnyRef]
+  override val JBoolean: ExpressionCompiler[JBoolean] = new ExpressionCompiler[JBoolean]
```

Comment:
I'm kind of surprised we need to be explicit about the types here (and other places) instead of them being inferred

Reply:
Right, I've had to set explicit types in a bunch of places otherwise the compiler complains about a type mismatch or an incompaible type

As an example his is from the Scala 3 migration guide: https://docs.scala-lang.org/scala3/guides/migration/incompat-type-inference.html

"In Scala 3 the return type of an override method is inferred by inheritance from the base method, whereas in Scala 2.13 it is inferred from the left hand side of the override method."

Reply:
I hate this crap. Java style being enforced on scala code. Having to state the type twice on the same line of code is well,... barfy java stuff. 

In this case, I suggest dropping the overrides by merging this object with the ExpressionCompilersClass base. That is used in exactly one place. Might as well just pass the object.

Reply:
I don't think it's at straightfoward as merging the object (which is in core) with the ExpressionCompilerClass base (which is in runtime1). ExpressionCompilerClass wont have access to the ExpressionCompiler class in core, unless you want me to merge the ExpressionCompiler class with it's own base?

Reply:
Ah. This stuff again. This split between runtime1 and core needs to be revisited at some point it's not a stable division of the modules. So I guess the repetitiveness in type signatures is needed because of the module split then. Oh well.